openedx/openedx-platform
# feat: add generate report button in admin

### Description:

This PR adds a new field to SurveyReport model and adds add a button in the django model admin to generate a new report.

SurveyReport model (https://github.com/openedx/edx-platform/pull/31183).
SurveyReport Admin Model (https://github.com/openedx/edx-platform/pull/31323).

ISSUE:
https://github.com/openedx/tcril-engineering/issues/349

APPROVED DESIGN:
https://docs.google.com/document/d/1tm1O5BQsAwekphI5IN9pBn8behQDZigXePLGqCl5he8

### DECISIONS:
- We decided add a new celery task to generate the report async, because we won't want block the user experience when this decide generate a new report and the generating time depends of how much information the user have.

### How to test
With a dev environment running, you should go to the django admin `/admin/survey_report/surveyreport/` and use the button "Generate Report" to generate a new survey report.

You should see a new report with state "Processing" if you reload the page so few seconds later you should see the same report with state "Generated" or "Error".
